src/capi.ts

```diff
@@ -102,6 +102,42 @@ const includesTweets = (content: Content): boolean => {
         .some(Boolean)
 }
 
+const includesInstagram = (content: Content): boolean => {
```

Comment:
I was thinking it could be a generic function with an argument of the element type? `ElementType.INSTAGRAM`

Reply:
To avoid iterating through the list multiple times you could do something like this:

```ts
type ThirdPartyEmbeds = {
  twitter: boolean;
  youtube: boolean;
  instagram: boolean;
  spotify: boolean;
};

const noThirdPartyEmbeds: ThirdPartyEmbeds = {
  twitter: false;
  youtube: false;
  instagram: false;
  spotify: false;
};

const checkForThirdPartyEmbed = (thirdPartyEmbeds: ThirdPartyEmbeds, element: BodyElement): ThirdPartyEmbeds => {

  switch (element.type) {

    case ElementType.INSTAGRAM:
        return { ...thirdPartyEmbeds, instagram: true };

    case ElementType.TWEET:
        return { ...thirdPartyEmbeds, twitter: true };

    case ElementType.VIDEO:
        return { ...thirdPartyEmbeds, youtube: true };

    case ElementType.AUDIO:
        return { ...thirdPartyEmbeds, spotify: true };

    default:
      return thirdPartEmbeds;

}

body.reduce(
  (thirdPartyEmbeds, block) => block.elements.reduce(checkForThirdPartyEmbed, thirdPartyEmbeds),
  noThirdPartyEmbeds,
);
```

with `checkForThirdPartyEmbed` playing the role of the `switch...case`/function @frankie297 and @webb04 mention above. Haven't tested that but I think it should work 🤔.

The `ThirdPartyEmbeds` object can then be used to reduce the number of arguments to the `buildCsp` function, as @frankie297 [mentions below](https://github.com/guardian/apps-rendering/pull/789/files#r500271885).

Reply:
Wow thats a great implementation, never thought of reduce as a solution here

Reply:
Thanks Jamie going to implement all these, maybe in incremental pushes in some cases, as they are quite involved👍